Cargo.toml

```diff
@@ -32,16 +32,16 @@ autobenches = false
 [dependencies]
 approx = "0.5"
 itertools = "0.10"
-nalgebra = { version = "0.29", default-features = false, features = ["rand", "std"] }
+nalgebra = { version = "0.31", default-features = false, features = ["rand", "std"] }
 num-traits = "0.2"
 rayon = "1.3"
 tracing = { version = "0.1", default-features = false, features = ["std"] }
 rand = "0.8"
 sprs = { version = "0.11", default-features = false, features = ["multi_thread"] }
 
 [dev-dependencies]
-criterion = "0.3"
+criterion = "0.4"
```

Comment:
Actually since 2947513b4a68015e82739b89adcab468faf4b63d coupe doesn't depend on criterion and this line can be removed.